src/osd/PG.cc

```diff
@@ -3855,9 +3882,25 @@ void PG::replica_scrub(
   if (!end.is_max())
     end.pool = info.pgid.pool();
 
-  build_scrub_map_chunk(
-    map, start, end, msg->deep, msg->seed,
-    handle);
+  if (scrubber.map_state == PG::Scrubber::REG_SCRUBBING) {
+    assert(scrubber.scrubmap.objects.empty());
+    assert(scrubber.ls.empty());
+    build_scrub_map_chunk(scrubber.scrubmap, start, end, msg->seed, handle, scrubber.ls);
+  } else {
+    assert(scrubber.map_state == PG::Scrubber::DEEP_SCRUBBING);
+    get_pgbackend()->be_deep_scan_list(scrubber.scrubmap, scrubber.ls, msg->seed, handle);
```

Comment:
It seems to me like you still want to use build_scrub_map_chunk.  Even with a deep scrub, we do want to verify the metadata stuff from build_scrub_map_chunk.